projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_dag.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import json
+import logging
+import os
+import pprint
+import sys
+from graphlib import TopologicalSorter
+from typing import Any
+from typing import Dict
+from typing import List
+
+from taurus_datajob_api import ApiException
+from vdk.plugin.meta_jobs.cached_data_job_executor import TrackingDataJobExecutor
+from vdk.plugin.meta_jobs.meta import TrackableJob
+from vdk.plugin.meta_jobs.remote_data_job_executor import RemoteDataJobExecutor
+from vdk.plugin.meta_jobs.time_based_queue import TimeBasedQueue
+
+log = logging.getLogger(__name__)
+
+
+class MetaJobsDag:
+    def __init__(self, team_name: str):
+        self._team_name = team_name
+        self._topological_sorter = TopologicalSorter()
+        self._delayed_starting_jobs = TimeBasedQueue(
+            min_ready_time_seconds=int(
+                os.environ.get("VDK_META_JOBS_DELAYED_JOBS_MIN_DELAY_SECONDS", 30)
+            ),
+            randomize_delay_seconds=int(
+                os.environ.get(
+                    "VDK_META_JOBS_DELAYED_JOBS_RANDOMIZED_ADDED_DELAY_SECONDS", 600
+                )
+            ),
+        )
+        self._finished_jobs = []
+        self._job_executor = TrackingDataJobExecutor(RemoteDataJobExecutor())
+
+    def build_dag(self, jobs: List[Dict]):
+        for job in jobs:
+            trackable_job = TrackableJob(
+                job["job_name"],
+                job.get("team_name", self._team_name),
+                job.get("fail_meta_job_on_error", True),
+            )
+            self._job_executor.register_job(trackable_job)
+            self._topological_sorter.add(trackable_job.job_name, *job["depends_on"])
+
+    def execute_dag(self):
+        self._topological_sorter.prepare()
+        while self._topological_sorter.is_active():
```

Comment:
Isn't this going to spin infinitely? Do we need some sleep() in this loop?

Reply:
Yes! I need to fix that.

Reply:
I'm sorry but did you fix this? I see you've marked the conversation as resolved...

Reply:
Hmm, not properly. Do unresolve a comment please when needed. 

I am not sure what is the appropriate etiquet in Github but I do use the resolved comment field to sort of tick that I've addressed a comment (be it with a comment response or a code change) but sometimes I am wrong.

Reply:
It's there actually - line 64 in meta_dag.py  

I added unit tests that proves that it's not going to spin infinitely (test_execute_dag_busyloop)